nixos/modules/services/hardware/logiops.nix

```diff
@@ -0,0 +1,164 @@
+                    mode: "OnRelease";
+                    action =
+                    {
+                      type: "Keypress";
+                      keys: ["KEY_UP"];
+                    };
+                  },
+                  {
+                    direction: "Down";
+                    mode: "OnRelease";
+                    action =
+                    {
+                      type: "Keypress";
+                      keys: ["KEY_DOWN"];
+                    };
+                  },
+                  {
+                    direction: "Left";
+                    mode: "OnRelease";
+                    action =
+                    {
+                      type: "CycleDPI";
+                      dpis: [400, 600, 800, 1000, 1200, 1400, 1600];
+                    };
+                  },
+                  {
+                    direction: "Right";
+                    mode: "OnRelease";
+                    action =
+                    {
+                      type = "ToggleSmartshift";
+                    }
+                  },
+                  {
+                    direction: "None"
+                    mode: "NoPress"
+                  }
+                );
+              };
+            },
+            {
+              cid: 0xc4;
+              action =
+              {
+                type: "Keypress";
+                keys: ["KEY_A"];
+              };
+            }
+          );
+        });
+      '';
+      description = mdDoc ''
+        Extra configuration that is appended to the end of the file.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.etc."logid.cfg".text = ''
+      ${cfg.extraConfig}
+    '';
+
+    systemd.services.logiops = {
+      description = "Logitech Configuration Daemon";
+      documentation = [ "https://github.com/PixlOne/logiops" ];
+
+      wantedBy = [ "graphical.target" ];
+
+      startLimitIntervalSec = 0;
+      after = [ "multi-user.target" ];
+      wants = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart = "${pkgs.logiops}/bin/logid";
+        Restart = "always";
+        # Ensures that the hid_logitech_hidpp module is loaded before logid starts
+        # https://github.com/PixlOne/logiops/issues/269#issuecomment-1012608838
+        ExecStartPre = "${pkgs.kmod}/bin/modprobe hid_logitech_hidpp || :";
+        User = "root";
+
+        RuntimeDirectory = "logiops";
+
+        CapabilityBoundingSet = [ "CAP_SYS_NICE" ];
+        DeviceAllow = [
+          "/dev/uinput rw"
+          # Globbing does not work here
+          # https://www.freedesktop.org/software/systemd/man/252/systemd.resource-control.html
+          "/dev/hidraw0 rw"
+          "/dev/hidraw1 rw"
+          "/dev/hidraw2 rw"
+          "/dev/hidraw3 rw"
+          "/dev/hidraw4 rw"
+          "/dev/hidraw5 rw"
+          "/dev/hidraw6 rw"
+          "/dev/hidraw7 rw"
+          "/dev/hidraw8 rw"
+          "/dev/hidraw9 rw"
+        ];
```

Comment:
This seems like it could break someones setup if they happened to have lots and lots of IO devices. Maybe it would be more reasonable to let the service have unrestricted access to devices, and rather accept the somewhat lower `systemd-analyze` score?

Reply:
Sorry, I'm not quite sure I understand how this could break someone's setup. As far as I can tell, the [`hid_logitech_hidpp` kernel module](https://github.com/torvalds/linux/blob/master/drivers/hid/hid-logitech-hidpp.c) only uses the `hidraw` devices.

Reply:
What happens if the connected mouse is `/dev/hidraw10`?

Reply:
Ah I see the issue now, thanks for the clarification. I'll make the change to allow permissions to all of `/dev`.